### DISCUSSION TITLE
`Allow GitHub Actions to create and approve pull requests` disabled

### DISCUSSION THREAD

**Discussion by bcldvd** [community/community#110415] [Actions]
### Select Topic Area

Bug

### Body

Hello everyone,

I'm trying to setup a release workflow so that it creates a pull request, in so doing I want to enable the `Allow GitHub Actions to create and approve pull requests` setting in my repository; however it is grayed out and having the weirdest behavior.

By default it's grayed out:
![image](https://github.com/community/community/assets/569053/29356b9e-9606-4f58-93a0-3e193a522d5d)

After toggling `Read repository contents and packages permissions`, and saving, I can now select it :

![image](https://github.com/community/community/assets/569053/09b8b2fd-18a9-49ae-9f37-87899dfd399c)

But here's the catch, when I save this; it reverts back to being disabled...

![image](https://github.com/community/community/assets/569053/63cc0032-507e-40e9-b1f6-eca270de5e7c)

Am I missing something ? Did anyone stumble upon this as well ?

**Comment by mrbalov** [comment]
I'm facing the same issue right now. And looks like this answers the question - https://stackoverflow.com/a/73155199/8097218.

**Reply by peterpeterparker** [reply]
Just faced the same issue, and it confirms what I always advise in terms of UX: users do not understand when things are simply disabled 😄. Thanks for sharing the solution link!

**Reply by yasuharu519** [reply]
I'm facing the same issue here. I followed the answers of the stackoverflow. I tried to change the setting of organization level of workflow permissions, but the checkbox is grayed out somehow in the organization setting, too. 😢

**Reply by yasuharu519** [reply]
Sorry, my procedure was wrong. First, I looked "organizational" level setting and I thought I cannot fix the issue. As an answer of stackoverflow says, we had to check setting of "enterprise level" through this address  https://github.com/enterprises/YOUR_ENTERPRISE/settings/actions 
Sorry for my misunderstanding.

**Comment by Lutherwaves** [comment]
I also saw the same issue, clicking the radio, saving, then refreshing makes the checkbox enabled

**Comment by WillNeill** [comment]
This is the classic enterprise-level permission override issue. The "Allow GitHub Actions to create and approve pull requests" checkbox is being blocked by a higher-level setting in your GitHub Enterprise/Organization.

GitHub has a three-tier permission hierarchy for Actions: Enterprise → Organization → Repository. A setting disabled at a higher tier cannot be enabled at a lower one — the checkbox will appear greyed out or silently revert. This catches a lot of people because the UI doesn't clearly explain why the setting won't stick.

How to fix it

1. Go to your Organization settings (not the repo): 
`https://github.com/organizations/[your-org]/settings/actions`
2. Scroll down to Workflow permissions
3. Make sure these are set:
```
Read and write permissions (radio button)
☑ Allow GitHub Actions to create and approve pull requests (checkbox)
```
4. Click Save
5. Then go back to the repo level:
https://github.com/[your-org]/[you-repo]/settings/actions
6. The checkbox should now be toggleable — enable it and save

Hope that helps anyone in future with this same issue.
